airflow/utils/log/file_task_handler.py

```diff
@@ -56,16 +55,19 @@ def close(self):
         if self.handler:
             self.handler.close()
 
-    def read(self, task_instance):
+    def read(self, task_instance, try_number):
```

Comment:
task_instance signature alone is definitely not enough, as we need to fetch logs from different previous try_number. Using `try_number=None` and return all logs separated by try_number sounds good.

Regarding the side note, try_number is definitely not ideal, as we need a better way to distinguish each task instance attempt. Right now `try_number` is used as a supplement to its primary key `dag_id`, `task_id` and `execution_date` to make each task attempt unique. In the future, we might need to change the db schema and use other fields as task_instance primary key.